### DISCUSSION TITLE
how to pass the uploaded filename to download event. the fileInput.filename is loat outside the upload event.

### DISCUSSION THREAD

**Discussion by bommithis** [pyscript/pyscript#1324] [Q&A]
def upload_file(event):
    if fileInput.value is not None:
        console.log(fileInput.filename)
        .... some code...

def get_file():
    some processing

file_download = pn.widgets.FileDownload(filename="file_name", callback=get_file, button_type="primary")

i want to use fileInput.filename from upload_file() for filename in file_download.

**Comment by JeffersGlass** [comment]
Hello! We would love to help you with this, but it's quite difficult to read your code. Would you edit your post (or reply to this one) with your code in a [code block](https://docs.pyscript.net/latest/reference/elements/py-config.html#supported-configuration-values) - that is, surround your code in three backticks like this:

\`\`\`python
def someCode():
   formatting?
\`\`\`

So that it shows up cleanly with indentation preserved:

```python
def someCode():
   formatting?
```
